From 333030adabd0cc6ac5546aac84cbafef70cf6d60 Mon Sep 17 00:00:00 2001
From: Sanyog Kale <skale@habana.ai>
Date: Wed, 13 Apr 2022 06:45:29 +0300
Subject: [PATCH] [SW-0] gaudi3: check image validation status in fit verify

This patch add checks in fit verification function and skip
verification if image validation is not enabled.

Change-Id: Ifafbff9466c85108e46f1610ac93d74fcb411984
---

diff --git a/arch/gaudi3/arc/secureboot.c b/arch/gaudi3/arc/secureboot.c
index b183d8d..30a4fe4 100644
--- a/arch/gaudi3/arc/secureboot.c
+++ b/arch/gaudi3/arc/secureboot.c
@@ -123,6 +123,9 @@
 	const void *pub_key_fit;
 	struct key_prop prop;
 
+	if (!security_efuse_image_validation_en())
+		return 0;
+
 	/* hl fit files are padded so the size is 4 bytes aligned */
 	img_size = ROUND_UP4(img_size);
 
